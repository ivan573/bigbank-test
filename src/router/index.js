import Vue from 'vue'
import VueRouter from 'vue-router'
import HomeView from '@/views/HomeView.vue'
import GameView from '@/views/GameView.vue'
import ShopView from '@/views/ShopView.vue'
import ReputationView from '@/views/ReputationView.vue'

if (process.env.NODE_ENV !== 'test') {
  Vue.use(VueRouter)
}

export const Route = {
  HOME: '/',
  GAME: '/game',
  SHOP: '/shop',
  REPUTATION: '/reputation'
}

export const routes = [
  {
    path: Route.HOME,
    name: 'home',
    component: HomeView
  },
  {
    path: Route.GAME,
    name: 'game',
    component: GameView
  },
  {
    path: Route.SHOP,
    name: 'shop',
    component: ShopView
  },
  {
    path: Route.REPUTATION,
    name: 'repotation',
    component: ReputationView
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

export default router
