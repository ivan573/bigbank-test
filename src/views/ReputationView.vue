<template>
 <div class="reputation">
   <h2 class="reputation__title">Your reputation</h2>
   <ul class="reputation__list">
    <li v-for="item in reputationArray" class="reputation__list-item" :key="item.name">
      {{ item.name }}: {{ item.value }}
    </li>
  </ul>
 </div>
</template>

<script>
import { mapState, mapGetters } from 'vuex'
import { Route } from '@/router/index'

export default {
  name: 'ReputationView',
  computed: {
    ...mapState('gameData', ['gameId']),
    ...mapGetters('reputation', ['reputationArray'])
  },
  created() {
    if (!this.gameId) {
      this.$router.push({ path: Route.HOME })
    }
  }
}
</script>

<style lang="scss" scoped>
  .reputation {
    margin: 0 auto;
    width: fit-content;

    &__list-item {
      text-transform: capitalize;
    }
  }
</style>
