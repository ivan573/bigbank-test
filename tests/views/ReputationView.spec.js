import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import { Route } from '@/router/index'
import ReputationView from '@/views/ReputationView.vue'

describe('ReputationView.vue', () => {
  const gameId = 'gameId'
  const reputationArray = [
    {
      name: 'people',
      value: 1.5
    },
    {
      name: 'state',
      value: 1
    },
    {
      name: 'underworld',
      value: 1
    }
  ]

  const wrapper = mount(ReputationView, {
    computed: {
      gameId() { return gameId },
      reputationArray() { return reputationArray }
    }
  })

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('displays reputation for each element from the array', () => {
    const reputationElements = wrapper.findAll('.reputation__list-item').wrappers

    expect(reputationElements.length).toBe(reputationArray.length)
    reputationElements.forEach((element, index) => {
      expect(element.text()).toBe(`${reputationArray[index].name}: ${reputationArray[index].value}`)
    })
  })

  it('redirects to root path if the game is not initialized', () => {
    const localVue = createLocalVue()
    localVue.use(VueRouter)

    const router = new VueRouter()
    router.push(Route.REPUTATION)

    const wrapper = shallowMount(ReputationView, {
      localVue,
      router,
      computed: {
        gameId() { return null },
        reputationArray() { return [] }
      }
    })

    expect(wrapper.vm.$route.path).toBe(Route.HOME)
  })
})
