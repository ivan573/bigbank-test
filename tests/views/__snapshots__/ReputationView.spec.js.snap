// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReputationView.vue matches snapshot 1`] = `
<div
  class="reputation"
>
  <h2
    class="reputation__title"
  >
    Your reputation
  </h2>
   
  <ul
    class="reputation__list"
  >
    <li
      class="reputation__list-item"
    >
      
     people: 1.5
   
    </li>
    <li
      class="reputation__list-item"
    >
      
     state: 1
   
    </li>
    <li
      class="reputation__list-item"
    >
      
     underworld: 1
   
    </li>
  </ul>
</div>
`;
